service/rtc/sfu.go

```diff
@@ -564,11 +608,17 @@ func (s *Server) InitSession(cfg SessionConfig, closeCb func() error) error {
 					)
 				}
 
-				if err := outScreenTrack.WriteRTP(packet); err != nil && !errors.Is(err, io.ErrClosedPipe) {
-					s.log.Error("failed to write RTP packet",
-						mlog.Err(err), mlog.String("sessionID", us.cfg.SessionID))
-					s.metrics.IncRTCErrors(us.cfg.GroupID, "rtp")
-					return
+				for i, writerCh := range writerChs {
+					// We need to copy the packet header to keep it race free in case
+					// of simulcast as we are dealing with concurrent writers.
+					pkt := *packet
+					pkt.Header = packet.Header.Clone()
+
+					select {
+					case writerCh <- &pkt:
+					default:
+						s.log.Error("failed to write RTP packet to writer channel", mlog.String("trackID", outScreenTracks[i].ID()))
```

Comment:
One unrelated positive thing about this approach is that if things start to go wrong the channel will become full and we'll log some errors. Before these changes, the process would simply struggle to send the packets in time but no error would show and I had to check on the client side for packet loss.